doc/query.md

```diff
@@ -39,25 +39,30 @@ Secondly, these documents are in the datastore:
 ```java
 { "name": "mike", 
   "age": 12, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Mike goes to middle school and likes reading books." };
 
 { "name": "mike", 
   "age": 34, 
-  "pet": {"species": "dog"} };
+  "pet": {"species": "dog"},
+  "comment": "Mike is a doctor and likes reading books." };
 
 { "name": "fred", 
   "age": 23, 
-  "pet": {"species": "cat"} };
+  "pet": {"species": "cat"},
+  "comment": "Fred works for a startup out of his home office." };
 ```
 
-### Creating indexes
+### Indexes
 
 In order to query documents, indexes need to be created over
 the fields to be queried against.
 
```

Comment:
Is the difference between a JSON index and a full text search index too obvious to explain? I think it would be somewhat useful/helpful to make a short, two to three sentence explanation between the two different types of indexes. You have already stated the example docs -- it would be easy to refer to them to say that a JSON-based query on "comment":"middle school" would not return any documents, but a full-text search on "comment":"$text":{"$search":"middle school"} would return the first document. I think the basic distinction is especially useful for new developers, but maybe that's just me.